packages/ui/components/ui/tx/view/swap/index.tsx

```diff
@@ -10,29 +10,16 @@ import {
   getClaimTx,
 } from '@penumbra-zone/getters/src/swap-view';
 import { ValueViewComponent } from '../value';
-import { ValueView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
-import { STAKING_TOKEN_METADATA } from '@penumbra-zone/constants/src/assets';
-import { Fee } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/fee/v1/fee_pb';
 import { ActionDetails } from '../action-details';
 import { UnimplementedView } from '../unimplemented-view';
-
-const getClaimFeeValueView = (claimFee: Fee) =>
-  new ValueView({
-    valueView: {
-      case: 'knownAssetId',
-      value: {
-        amount: claimFee.amount,
-        metadata: STAKING_TOKEN_METADATA,
-      },
-    },
-  });
```

Comment:
I believe that the ui package should not depend on `registry`. And also we should not pass additional data for rendering transaction view components   
In this case I had to simplify the fee display and get rid of ValueView

Reply:
Hmm, yeah, this is tricky. Can't think of a better solution for this either.

Reply:
it looks like there is no metadata field on the Fee type. we are using tools from `@penumbra-zone/perspective` to prepare the view prop everywhere we use this component, including a metadata getter. so i think options are

1. pr core for a metadata field on Fee, and prepare it before using component, as now
2. add a function prop to get metadata, component now responsible for all metadata
3. add a static optional prop for staking token metadata, component nearly identical

i prefer 3 because staking token metadata is a chain-level special case, and it's minimal change. the current behavior in this pr is good and should remain as fallback if prop is undefined/absent.

this can be a separate issue not addressed in this pr.

Reply:
Yeah, definitely doesn't need to be addressed as part of this PR

Reply:
Re: #1: I actually suspect that a better solution would be to represent the fee as a `ValueView` generated from the `Fee`, since `Fee` already contains both an amount and an asset ID. So e.g., in addition to the `swapPlaintext.claimFee` field, there could also be a `swapPlaintext.claimFeeView` field that's a `ValueView`. Thoughts?

(Again, though, out of scope for this PR.)